target/linux/ath79/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac

```diff
@@ -18,6 +18,12 @@ case "$board" in
 		[ "$PHYNBR" -eq 1 ] && \
 			echo $(k2t_get_mac "lan_mac") > /sys${DEVPATH}/macaddress
 		;;
+	dlink,dir-825-b1)
```

Comment:
please patch the mac address of the firmware extracted from flash. The k2t board already does it the wrong way. Please have a look at https://github.com/openwrt/openwrt/blob/master/target/linux/lantiq/base-files/etc/hotplug.d/firmware/12-ath9k-eeprom for examples.

Please don't use hardcoded mtdblocks. Use our helper functions like mtd_get_mac* or find_mtd_chardev. I've no idea how to content of mtd6 looks like, but something like mtd_get_mac_ascii might work.

Reply:
The problem that all existing functions won't work. I looked already there.
Hardcoded mtdblock is not good, it was a temp solution. I couldn't figure out why the script didn't work after sysupgrade without firstboot. I'll fix it.

Mac address is an ascii with colons. That mtd_get_mac_ascii requires some $key. I got a fixed addresses for both macs. I can write another proc or something else. There are many ways of getting it. The current is not good at all. But it works.

Reply:
It is easy `ath9k-eeprom-extract` will do it perfectly`.

Reply:
But we need to move it to functions or somewhere we can include it from.
I need it in 3 scripts.